### PR TITLE
feat: 414068 - Clear dog breaches when status is not In Breach

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/export/csv.js [hunk 1]
```diff
@@ -45,7 +45,8 @@ const headerRow = [
   'Withdrawn',
   'ExaminedByDlo',
   'MicrochipDeadline',
-  'NeuteringDeadline'
+  'NeuteringDeadline',
+  'InBreachReasons'
```
Comment (jbarnsley10) [inline app/export/csv.js:49]: I thought we said this new column would sit just after 'status'

File: app/overnight/expired-insurance.js [hunk 2]
```diff
@@ -16,24 +16,38 @@ const setExpiredInsuranceToBreach = async (today, user, t) => {
...
+        }
+      ],
       transaction: t
     })
+    const dogService = ServiceProvider.getDogService()
 
     for (const toUpdate of setToBreach) {
-      console.log(`Updating dog ${toUpdate.dog.index_number} to In breach`)
-      await updateStatusOnly(toUpdate.dog, statuses.InBreach, user, t)
+      await dogService.setBreach(toUpdate.dog, [breachReasons.NOT_COVERED_BY_INSURANCE], user, t)
```
Comment (jbarnsley10) [inline app/overnight/expired-insurance.js:50]: I'd like to keep the console.log in place as it's useful for logging the overnight processing

Comment (jbarnsley10) [inline app/overnight/expired-insurance.js:50]: Surely this should be set to INSURANCE_EXPIRED

File: app/repos/breaches.js [hunk 3]
```diff
@@ -11,14 +12,23 @@ const { BreachCategory } = require('../data/domain')
  *
  * @return {Promise<BreachCategoryDao[]>}
  */
-const getBreachCategoryDAOs = async () => {
+const getBreachCategoryDAOs = async (userSelectableOnly = false) => {
+  const where = userSelectableOnly
```
Comment (jbarnsley10) [inline app/repos/breaches.js:16]: What was the reasoning for a 'user-selectable' column?

File: app/service/dog.js [hunk 4]
```diff
@@ -1,46 +1,70 @@
...
+  /**
+   *
+   * @param {Dog} dog
+   * @param {string[]} dogBreaches
+   * @param {BreachCategory[]} allDogBreaches
+   * @param user
+   * @private
+   */
+  _prepareBreaches (dog, dogBreaches, allDogBreaches, user) {
     const preAuditDog = {
```
Comment (jbarnsley10) [inline app/service/dog.js:21]: Will the preAuditDog always be blank in terms of breaches? Could we have a situation where a dog's breaches change from one list to another?

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
Nice work. A couple of minor changes

File: app/overnight/expired-insurance.js (hunk 2 above)
Comment (whitewaterdesign) [reply app/overnight/expired-insurance.js:50]: very true 👍 good spot, thanks

Comment (whitewaterdesign) [reply app/overnight/expired-insurance.js:50]: my bad, was removed by mistake

File: app/service/dog.js (hunk 4 above)
Comment (whitewaterdesign) [reply app/service/dog.js:21]: this was the reason for our call last week and the question to Ben.  Theoretically if someone calls this endpoint with an In Breach dog the api allows you to set to In Breach, but update the breach reasons.  When that happens the audit log shows only changed fields.  As it was decided that the audit log should only show as describe (set to In breach, with the reasons), this approach creates the audit log in that fashion.

File: app/repos/breaches.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/repos/breaches.js:16]: otherwise the two new fields would show on the multiple select page

**Review by jbarnsley10 — APPROVED** [review]
Great

**Pull request merged**